staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -443,3 +444,19 @@ func CloneHeader(in http.Header) http.Header {
 	}
 	return out
 }
+
+// NewDefaultTransport returns a new instance of http.DefaultTransport
+func NewDefaultTransport() *http.Transport {
+	return &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
```

Comment:
These are just copied from the `http.DefaultTransport` of the go sdk.
Are those fine?
As far as i can see the following requests will be affected:
- Kube API server -> Kubelet (Only when using the SSH tunnel?)
- Kube API server -> Extension API servers (Always as far as i can see)

Reply:
Can and should we shallow copy the DefaultTransport first?

Reply:
Does it even differ from `http.DefaultTransport`?

Reply:
I didn't do this for 2 reasons:
- https://github.com/golang/go/issues/26013#issuecomment-399481302
  > Well, it is safe to do this: ... shallow copy example ... but only early in the program before the DefaultTransport has been used.

  I wasn't sure if we can guarantee this
- When updating the used Go version we might unintentionally change the behavior of used http clients?